packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -167,7 +167,7 @@ export async function _receiveCloudEventsDeserialize(
         id: p.event["id"],
         source: p.event["source"],
         data: p.event["data"],
-        dataBase64: p.event["data_base64"],
+        dataBase64: Buffer.from(p.event["data_base64"] ?? ""),
```

Comment:
As in the RLC output, we generate byte-array as string, and in the modular layer we generate it as Uint8Array, we need to use Buffer.from to convert the type.

Reply:
Isn't Buffer.from only supported in Node but not in browser?

Reply:
Yeah I think we need to have some environment specific handling here. Found some prior art in keyvault :

https://github.com/Azure/azure-sdk-for-js/blob/9c9c709d5827b28ebdb9d09c2c1c82fe46d23442/sdk/keyvault/keyvault-certificates/src/utils.ts#L35

As well as special handling for base64 specificaly in core-client:
https://github.com/Azure/azure-sdk-for-js/blob/9c9c709d5827b28ebdb9d09c2c1c82fe46d23442/sdk/core/core-client/src/base64.browser.ts
https://github.com/Azure/azure-sdk-for-js/blob/9c9c709d5827b28ebdb9d09c2c1c82fe46d23442/sdk/core/core-client/src/base64.ts

Reply:
@xirzec Thank you for the pointers, I will take a look and see if it's necessary to file another PR to fix this issue.

Reply:
I'm fine fixing this in a follow up pr

Reply:
Thank you :)